tests/framework/PostProcessors/SparseSensing/tests

```diff
@@ -0,0 +1,8 @@
+[Tests]
+  [./testSPSLOptiTwist]
+    type = 'RavenFramework'
+    input = 'testSPSLOptiTwist.xml'
+    csv = 'reconstructionOptiTwist/outPP.csv'
+    output = 'reconstructionOptiTwist/mySensorPlot_scatter-scatter.png'
```

Comment:
@joshua-cogliati-inl, could you please modify this to test the png (compare with gold)? In addition, it seems it plots the last sample or maybe all of them on top of each other. Can we add user input to require which temperature (which sample) to be plotted? or at least plot the reference one?

Reply:
FYI, if test the png, then we will only test when we test optional packages.

Reply:
Hm, plotting a specific sample may be hard with the way the data is arranged.  For example, I tried:
```diff
diff --git a/tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml 
b/tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml
index aa29c90d7..f8c28c01f 100644
--- a/tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml
+++ b/tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml
@@ -10,7 +10,7 @@
 
   <RunInfo>
     <WorkingDir>reconstructionOptiTwist</WorkingDir>
-    <Sequence>LoadCSV, mySPpp, print</Sequence>
+    <Sequence>LoadCSV, mySPpp,avetemp, print</Sequence>
     <batchSize>1</batchSize>
   </RunInfo>
 
@@ -28,10 +28,18 @@
       <Model class="Models" type="PostProcessor">mySPSL</Model>
       <Output class="DataObjects" type="DataSet">outPP</Output>
     </PostProcess>
+    <PostProcess name="avetemp">
+      <Input class="DataObjects" type="HistorySet">TwistDO</Input>
+      <Model class="Models" type="PostProcessor">average_temp</Model>
+      <Output class="DataObjects" type="PointSet">TwistDOTempAve</Output>
+    </PostProcess>
     <IOStep name="print">
       <Input class="DataObjects" type="DataSet">outPP</Input>
       <Input class="DataObjects" type="HistorySet">TwistDO</Input>
+      <Input class="DataObjects" type="PointSet">TwistDOTempAve</Input>
       <Output class="OutStreams" type="Print">outPP</Output>
+      <Output class="OutStreams" type="Print">outTwistDO</Output>
+      <Output class="OutStreams" type="Print">outTwistDOTempAve</Output>
       <Output class="OutStreams" type="Plot">mySensorPlot</Output>
     </IOStep>
   </Steps>
@@ -48,6 +56,10 @@
         <optimizer>QR</optimizer>
       </Goal>
     </PostProcessor>
+    <PostProcessor name="average_temp" subType="HStoPSOperator">
+      <pivotParameter>BCTemperature (K)</pivotParameter>
+      <operator>average</operator>
+    </PostProcessor>
   </Models>
 
   <DataObjects>
@@ -58,6 +70,10 @@
         <pivotParameter>index</pivotParameter>
       </options>
     </HistorySet>
+    <PointSet name="TwistDOTempAve">
+      <Input>Power (W)</Input>
+      <Output>X (m),Temperature (K),Y (m)</Output>
+    </PointSet>
     <DataSet name="outPP">
       <Input>InputPlaceHolder</Input>
       <Output>Temperature (K), X (m), Y (m)</Output>
@@ -71,6 +87,14 @@
       <source>outPP</source>
       <what>output</what>
     </Print>
+    <Print name="outTwistDO">
+      <type>csv</type>
+      <source>TwistDO</source>
+    </Print>
+    <Print name="outTwistDOTempAve">
+      <type>csv</type>
+      <source>TwistDOTempAve</source>
+    </Print>
     <Plot name="mySensorPlot">
       <plotSettings>
         <plot>
```
but that resulted in:
```
Power (W),X (m),Temperature (K),Y (m)
350,0.0107616743047,241.831291133,0.08204888554
350,0.0107616743047,273.515621667,0.08204888554
350,0.0107616743047,303.282704721,0.08204888554
350,0.0107616743047,332.908743084,0.08204888554
```
which drops the index. I am not sure there is a quick fix for this.

Reply:
If we repeat the test multiple times the way it was, is the color map consistent? which temp is it showing? all on top of each other, or is it the first, or last? can you check that?

Reply:
Based on the color map, it is showing the 330 K BC temperature data.

Reply:
![3dsensorplot](https://github.com/idaholab/raven/assets/7118148/2fad3451-e9a5-42e9-84d8-42c28669e14d)
Here is a plot with:
```
        <plot>
          <type>scatter</type>
          <x>TwistDO|Output|X (m)</x>
          <y>TwistDO|Output|Y (m)</y>
          <z>TwistDO|Input|BCTemperature (K)</z>
          <cmap>coolwarm</cmap>
          <colorMap>TwistDO|Output|Temperature (K)</colorMap>
        </plot>
```
So the data for all four temperatures is there, but just the highest boundary condition is drawn.